src/cmd/manage_nsfs.js

```diff
@@ -395,111 +389,75 @@ async function fetch_existing_account_data(action, target, decrypt_secret_key) {
-
+/**
+ * update_account does the following updates an existing config file, 
+ * if a new_name provided it will pass old_name to the update function for unlinking the index
+ * if a new_access_key provided it will pass the cur_access_key for unlinking the index
+ * @param {Object} data
+ * @returns { Promise<{ code: typeof ManageCLIResponse.AccountUpdated, detail: Object }>} 
+ */
+async function update_account(data) {
     const cur_name = data.name;
     const new_name = data.new_name;
     const cur_access_key = has_access_keys(data.access_keys) ? data.access_keys[0].access_key : undefined;
-    const update_name = new_name && cur_name && data.new_name !== cur_name;
-    const update_access_key = data.new_access_key && cur_access_key && data.new_access_key.unwrap() !== cur_access_key.unwrap();
-
-    if (!update_name && !update_access_key) {
-        if (data.new_access_key) {
-            // the user set the same access-key as was before
-            data.access_keys[0] = _.pick(data.access_keys[0], ['access_key', 'secret_key']);
-            data = _.omit(data, ['new_access_key']);
-        }
-        const encrypted_account = await nc_mkm.encrypt_access_keys(data);
-        data.master_key_id = encrypted_account.master_key_id;
-        const encrypted_data = JSON.stringify(encrypted_account);
-        data = _.omitBy(data, _.isUndefined);
-        // We take an object that was stringify
-        // (it unwraps ths sensitive strings, creation_date to string and removes undefined parameters)
-        // for validating against the schema we need an object, hence we parse it back to object
-        const account = encrypted_data ? JSON.parse(encrypted_data) : data;
-        nsfs_schema_utils.validate_account_schema(account);
-        await config_fs.update_account_config_file(account);
-        write_stdout_response(ManageCLIResponse.AccountUpdated, data);
-        return;
+    const update_name = is_name_update(data);
+    const update_access_key = is_access_key_update(data);
+
+    const account_name = new_name || cur_name;
+    data.name = account_name;
+    data.email = account_name; // saved internally
+    if (data.new_access_key) {
+        data.access_keys[0] = {
+            access_key: data.new_access_key,
+            secret_key: data.access_keys[0].secret_key,
+        };
     }
-    const data_name = new_name || cur_name;
-    data.name = data_name;
-    data.email = data_name; // saved internally
-    data.access_keys[0] = {
-        access_key: data.new_access_key || cur_access_key,
-        secret_key: data.access_keys[0].secret_key,
-    };
-
-    const name_exists = update_name && await config_fs.is_account_exists_by_name(data.name, undefined);
-    const access_key_exists = update_access_key &&
-        await config_fs.is_account_exists_by_access_key(data.access_keys[0].access_key.unwrap());
+    const cli_flag_to_remove = ['new_name', 'new_access_key'];
```

Comment:
To match what written in buckets:
```suggestion
    const cli_account_flags_to_remove = ['new_name', 'new_access_key'];
```